.github/workflows/checkstyle.yml

```diff
@@ -0,0 +1,23 @@
+name: Java CI with Maven for checkstyle plugin
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
```

Comment:
@vpetrovaa using latest it is not good

also, consider running this check on multiples OSs, like macOS, windows, etc.